app/init.php

```diff
@@ -1622,6 +1624,76 @@ function getDevice($root): Device
     return $register->get('geodb');
 }, ['register']);
 
+App::setResource('ip', function ($request) {
+    return $request->getIp();
+}, ['request']);
+
+App::setResource('geodbClient', function () {
+    $client = new Client();
+    $client->addHeader('Authorization', 'Bearer ' . System::getEnv('_APP_GEO_SECRET'));
+    $client->setBaseUrl('http://appwrite-geo/v1');
+    return $client;
+});
+
+App::setResource('geoRecordDocker', function ($ip, $geodbClient) {
+    $record = null;
+    try {
+        $record = $geodbClient->fetch("/ips/{$ip}", Client::METHOD_GET);
+        $record = $record->json();
+    } catch (Throwable $th) {
+        Console::error($th->getMessage());
+        Console::error($th->getTraceAsString());
+        $record = null;
+    }
+
+    return $record;
+});
+
+App::setResource('geoRecordEmbeded', function ($ip, $geodb) {
```

Comment:
Docker will always be there, right? Can we not have Embeded version anymore?

Reply:
We want to fallback in case the docker fails due to any issues like network, connection or other unknown issues.